src/third_party/dartino/object_memory.cc

```diff
@@ -212,17 +212,31 @@ void Chunk::find(uword word, const char* name) {
 #endif
 
 Chunk* ObjectMemory::allocate_chunk(Space* owner, uword size) {
+  static const int UNUSABLE_SIZE = 50;
+  void* unusable_pages[UNUSABLE_SIZE];
+  int unusable_pages_count = 0;
   size = Utils::round_up(size, TOIT_PAGE_SIZE);
   void* memory = OS::allocate_pages(size);
   uword lowest = GcMetadata::lowest_old_space_address();
   USE(lowest);
-  if (memory == null) return null;
-  if (reinterpret_cast<uword>(memory) < lowest ||
-      reinterpret_cast<uword>(memory) - lowest + size > GcMetadata::heap_extent()) {
-    printf("New allocation %p-%p\n", memory, unvoid_cast<char*>(memory) + size);
-    FATAL("Toit heap outside expected range");
+  while (memory != null &&
+         (reinterpret_cast<uword>(memory) < lowest ||
+          reinterpret_cast<uword>(memory) - lowest + size > GcMetadata::heap_extent())) {
```

Comment:
Maybe add a helper that determines if a memory pointer is in the old space range?